build/ci/Makefile

```diff
@@ -171,6 +169,12 @@ ci-gke-cleanup: ci-e2e-delete-cluster
     	-e "GCLOUD_PROJECT=$(GCLOUD_PROJECT)" \
     	-e "GKE_CLUSTER_NAME=$(GKE_CLUSTER_NAME)" \
     	-e "GKE_SERVICE_ACCOUNT_KEY_FILE=$(GO_MOUNT_PATH)/build/ci/$(GKE_CREDS_FILE)" \
-    	cloud-on-k8s-ci-e2e \
+    	$(IMAGE) \
     	bash -c "GKE_CLUSTER_VERSION=1.11 $(GO_MOUNT_PATH)/operators/hack/gke-cluster.sh auth && \
     	 	$(GO_MOUNT_PATH)/build/ci/delete_unused_disks.py"
+
+# Check if Docker image exists by trying to pull it. If there is no image, then build and push it.
+ci-build-image: vault-docker-creds
+	@ docker pull $(IMAGE) || (docker build -f $(ROOT_DIR)/build/ci/Dockerfile -t $(IMAGE) $(ROOT_DIR) &&\
```

Comment:
Would it be helpful to add a [label](https://docs.docker.com/engine/reference/builder/#label) to the Docker image to show the current commit hash?.  This is because now the image tag is a combined hash of all the files and it's not quite straightforward to figure out when a particular image was constructed.

Something like:

```Makefile
...
docker build -f $(ROOT_DIR)/build/ci/Dockerfile -t $(IMAGE) --label "commit.hash=$(TAG)" $(ROOT_DIR)
...
```

Reply:
Added